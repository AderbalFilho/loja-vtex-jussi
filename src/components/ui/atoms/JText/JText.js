import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './JText.scss';

function JText({ children, classes, color, size, ...rest }) {
  const className = classNames(
    'j-text',
    `j-text--${color}`,
    `j-text--${size}`,
    classes
  );

  return (
    <p className={className} {...rest}>
      {children}
    </p>
  );
}

JText.defaultProps = {
  size: 'normal',
  color: 'default',
};

JText.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  size: PropTypes.oneOf(['normal', 'small']),
  color: PropTypes.oneOf(['default', 'white', 'black', 'dark-grey', 'pink']),
};

export default JText;
