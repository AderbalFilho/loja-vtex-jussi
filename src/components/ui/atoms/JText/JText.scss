@import 'styles/mixin';

.j-text {
  @extend %text-paragraph;
}

.j-text--small {
  font-size: 0.75rem;
  line-height: 1.125rem;
}

.j-text--white {
  color: var(--white);
}

.j-text--black {
  color: var(--black);
}

.j-text--dark-grey {
  color: var(--dark-grey);
}

.j-text--pink {
  color: var(--pink-jussi);
}
