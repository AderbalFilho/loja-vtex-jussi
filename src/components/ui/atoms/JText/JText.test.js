import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import JText from './JText';

describe('JText', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<JText>Jüssi</JText>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  describe('change size', () => {
    it('renders size normal when no size is passed', () => {
      const { container } = render(<JText>Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--normal');
    });

    it('renders size small', () => {
      const { container } = render(<JText size="small">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--small');
    });

    it('renders size normal', () => {
      const { container } = render(<JText size="normal">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--normal');
    });
  });

  describe('change color', () => {
    it('renders color default when no color is passed', () => {
      const { container } = render(<JText>Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--default');
    });

    it('renders color white', () => {
      const { container } = render(<JText color="white">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--white');
    });

    it('renders color black', () => {
      const { container } = render(<JText color="black">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--black');
    });

    it('renders color dark-grey', () => {
      const { container } = render(<JText color="dark-grey">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--dark-grey');
    });

    it('renders color pink', () => {
      const { container } = render(<JText color="pink">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--pink');
    });

    it('renders color default', () => {
      const { container } = render(<JText color="default">Jüssi</JText>);
      const text = container.querySelector('.j-text');

      expect(text.classList).toContain('j-text--default');
    });
  });

  it('renders children', () => {
    const { container } = render(<JText>Jüssi</JText>);
    const text = container.querySelector('.j-text');

    expect(text.textContent).toBe('Jüssi');
  });

  it('renders classes', () => {
    const { container } = render(<JText classes="text1">Jüssi</JText>);
    const text = container.querySelector('.j-text');

    expect(text.classList).toContain('text1');
  });
});
