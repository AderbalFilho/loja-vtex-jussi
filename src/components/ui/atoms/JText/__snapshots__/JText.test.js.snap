// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JText matches snapshot 1`] = `
<p
  className="j-text j-text--default j-text--normal"
>
  JÃ¼ssi
</p>
`;
